src/behavioural-events/events.ts

```diff
@@ -0,0 +1,46 @@
+import { v4 as uuidv4 } from "uuid";
+import { type Trace } from "./trace";
+
+type EventType = "rcb_sdk_initialized";
+
+export abstract class BaseEvent {
+  public readonly id: string;
+  public readonly timestamp: number;
+  public readonly type: string;
+  public readonly traceId: string;
+  public readonly traceIndex: number;
```

Comment:
Sending the `traceIndex` in case some manipulation of the timestamp for partial ordering events in the same trace is needed.